simpleast-core/src/test/java/com/discord/simpleast/code/JavaScriptRuleTest.kt

```diff
@@ -0,0 +1,224 @@
+        "'Hello'",
+        "\"world\"",
+        "`Hi`")
+  }
+  @Test
+  fun stringsMultiline() {
+    val ast = parser.parse("""
+      ```js
+      const text = `
+      hello
+      world
+      `;
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<StyleNode.TextStyledNode<*>>(
+        "`",
+        """
+          hello
+          world
+        """.trimIndent(),
+        "`"
+    )
+  }
+
+  @Test
+  fun functions() {
+    val ast = parser.parse("""
+      ```js
+      function test(T) {
+        // Implementation
+      }
+      const fn = function() {};
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<JavaScript.FunctionNode<*>>(
+      """function test(T) {
+           // Implementation
+         }""".trimIndent(),
+      "function() {}")
+  }
+
+  @Test
+  fun commentedFunction() {
+    val ast = parser.parse("""
+      ```js
+      /*
+        function test(T) {
+          throw new Error();
+        }
+      */
+      // function O() {}
+      console.log(x /* test var */);
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<StyleNode.TextStyledNode<*>>(
+        """
+        /*
+          function test(T) {
+            throw new Error();
+          }
+        */
+        """.trimIndent(),
+        "// function O() {}",
+        "/* test var */")
+  }
+
+  @Test
+  fun keywords() {
+    val ast = parser.parse("""
+      ```js
+      const x = 'Test';
+      while (true) {}
+      for (;;) {}
+      if (false) {}
+      class Test {}
+      try {
+        // Nothing
+      } catch (err) {
+        console.error(err);
+      } finally {
+        return;
+      }
+      ```
+    """.trimIndent(), TestState())
+    ast.assertNodeContents<StyleNode.TextStyledNode<*>>(
+        "const", "while", "true", 
+        "for", "if", "false",
+        "class", "try", "catch",
+        "finally", "return")
+  }
+
+  @Test
+  fun numbers() {
+    val ast = parser.parse("""
+      ```js
+      let x = 0;
+      x += 69;
+      Math.max(x, 420);
+      ```
+    """.trimIndent(), TestState())
+    ast.assertNodeContents<StyleNode.TextStyledNode<*>>(
+            "0", "69", "420"
+   )
+  }
+
+  @Test
+  fun fields() {
+    val ast = parser.parse("""
+      ```js
+      var x = 10;
+      let foo = 'bar';
+      const baz = foo;
+        ```
+    """.trimIndent(), TestState())
+    ast.assertNodeContents<JavaScript.FieldNode<*>>(
+        "var x",
+        "let foo",
+        "const baz")
+  }
+
+  @Test
+  fun classDef() {
+    val ast = parser.parse("""
+      ```js
+      class Bug extends Error {}
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<CodeNode.DefinitionNode<*>>(
+        "class Bug")
+  }
+
+  @Test
+  fun regex() {
+    val ast = parser.parse("""
+      ```js
+      /(.*)/g
+      /[\$\{\}]/
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<StyleNode.TextStyledNode<*>>("/(.*)/g", "/[\$\{\}]/")
```

Comment:
```suggestion
    ast.assertNodeContents<StyleNode.TextStyledNode<*>>("/(.*)/g", """/[\$\{\}]/""")
```
Conver this to a raw string because of the regex